wp-parsely.php

```diff
@@ -47,12 +226,44 @@ function maybe_load_plugin() {
 	 *
 	 * add_filter( 'wpvip_parsely_load_mu', '__return_false' );
 	 */
-	if ( ! apply_filters( 'wpvip_parsely_load_mu', get_option( '_wpvip_parsely_mu' ) === '1' ) ) {
-		return;
-	}
+	$do_not_load_plugin = ! apply_filters( 'wpvip_parsely_load_mu', get_option( '_wpvip_parsely_mu' ) === '1' );
+
+	// Check if wp-parsely has already been loaded, for example when the user
+	// has a self-managed wp-parsely installation.
+	$plugin_is_already_loaded = in_array( PARSELY_PLUGIN_SIGNATURE, get_option( 'active_plugins' ) );
+
+	// Check if wp-parsely is in a state of attempting to be activated
+	// for a self-managed wp-parsely installation.
+	$plugin_loading_queued = is_queued_for_activation();
+
+	// If the plugin must not be loaded, set Loader Info data and exit.
+	if ( $do_not_load_plugin || $plugin_is_already_loaded || $plugin_loading_queued ) {
+
+		// Attempt to detect force-disables invoked by the wpvip_parsely_load_mu filter.
+		// Set Loader Info data as an inactive mu-plugins integration.
+		$force_disabled = $do_not_load_plugin && has_filter( 'wpvip_parsely_load_mu' )
+			&& false === $plugin_is_already_loaded && false === $plugin_loading_queued;
+		if ( $force_disabled ) {
+			Parsely_Loader_Info::set_active( false );
+			Parsely_Loader_Info::set_integration_type( Parsely_Loader_Info::INTEGRATION_TYPE_MUPLUGINS );
+
+			if ( get_option( '_wpvip_parsely_mu' ) === '1' ) {
+				Parsely_Loader_Info::set_integration_type( Parsely_Loader_Info::INTEGRATION_TYPE_MUPLUGINS_SILENT );
+			}
+		}
+
+		// Set Loader Info data as an active self-managed integration.
+		if ( $plugin_is_already_loaded || $plugin_loading_queued ) {
+			Parsely_Loader_Info::set_active( true );
+			Parsely_Loader_Info::set_integration_type( Parsely_Loader_Info::INTEGRATION_TYPE_SELF_MANAGED );
+
+			//Get the version of the self-managed wp-parsely plugin.
+			$parsely_options = Parsely_Loader_Info::get_parsely_options();
+			if ( array_key_exists( 'plugin_version', $parsely_options ) ) {
+				Parsely_Loader_Info::set_version( $parsely_options['plugin_version'] );
+			}
+		}
```

Comment:
Question: I'm wondering whether we want to run this during plugin activation? Opinions?

Reply:
I don't have strong opinions on this. I've removed it in the latest commit.